pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -801,6 +826,53 @@ func (c *Cluster) getExpandPVCInitContainer(osdProps osdProperties, osdID string
 		SecurityContext: PrivilegedContext(),
 		Resources:       osdProps.resources,
 	}
+}
 
-	return container
+func (c *Cluster) getExpandEncryptedPVCInitContainer(mountPath string, osdProps osdProperties) v1.Container {
+	/* Command example
+	   [root@rook-ceph-osd-0-59b9947547-w8mdq /]# cryptsetup resize set1-data-2-8n462-block-dmcrypt
+	   Command successful.
+	*/
+
+	return v1.Container{
+		Name:  expandEncryptedPVCOSDInitContainer,
+		Image: c.spec.CephVersion.Image,
+		Command: []string{
+			"cryptsetup",
+		},
+		Args:            []string{"--verbose", "resize", encryptionDMName(osdProps.pvc.ClaimName, DmcryptBlockType)},
+		VolumeMounts:    []v1.VolumeMount{getPvcOSDBridgeMountActivate(mountPath, osdProps.pvc.ClaimName)},
+		SecurityContext: PrivilegedContext(),
+		Resources:       osdProps.resources,
+	}
+}
+
+func (c *Cluster) getEncryptedStatusPVCInitContainer(mountPath string, osdProps osdProperties) v1.Container {
```

Comment:
Why do we need this init container? Is it just to log the status? If it's just for debugging, doesn't seem like we should have it in production.

Reply:
I think this is useful to validate a few things, such as the encryption expansion has worked correctly. I think it's a nice to have, not necessarily a must-have though.